docs/design/api/storage.rst

```diff
@@ -0,0 +1,324 @@
+Storage API
+###########
+
+For this examples let's say:
+
+- Buyers logs with Firefox Account Bearer Tokens
+- Seller apps logs with Hawk
+- Payment App logs with Basic Auth
+
+
+As a user, I want to retrieve my information about my purchases
+===============================================================
+
+.. code-block:: http
+
+    GET /buckets/mozilla/collections/payments/records
+    Authorization: Bearer <User Firefox Account Bearer Token>
+
+
+    HTTP/1.1 200 OK
+    Content-Type: application/json; charset=UTF-8
+
+    {
+      "items": [
+        {
+            "id": "dc86afa9-a839-4ce1-ae02-3d538b75496f",
+            "last_modified": 1432024555580,
+            "buyer": "fxa:5331be33303ccff19d3c49b6da276913",
+            "seller": "hawk:144c22a77d75937740ec3c957fbdb1d1",
+            "product": "bb5bf35f-cb2b-40e7-b1ef-8b097bd550f2",
+            "added_on": 1432024555580,
+            "valid_until": 1463560555580
+        },
+        {
+            "id": "23160c47-27a5-41f6-9164-21d46141804d",
+            "last_modified": 1430140411480,
+            "title": "MoFo",
+            "buyer": "fxa:5331be33303ccff19d3c49b6da276913",
+            "seller": "hawk:5851128c0c7d72c6845339599b46d092",
+            "product": "bda9d953-dd81-427d-b7fc-7bcb0df9d666",
+            "added_on": 1430140411480,
+            "valid_until": 1432732411480
+        }
+      ]
+    }
+
+
+As the payments app, I want to submit new payment information for a given user
+==============================================================================
+
+.. code-block:: http
+
+    POST /buckets/mozilla/collections/payments/records
+    Authorization: Basic <PaimentApp Basic Auth Credentials>
+
+    {
+      "data": {
+        "last_modified": 1432024555580,
+        "buyer": "fxa:5331be33303ccff19d3c49b6da276913",
+        "seller": "hawk:144c22a77d75937740ec3c957fbdb1d1",
+        "product": "bb5bf35f-cb2b-40e7-b1ef-8b097bd550f2",
+        "added_on": 1432024555580,
```

Comment:
I wonder if we need a start date for a product. I can't really think of a realistic case for that. We could always add it later if needed.